pkg/oci/config_validate.go

```diff
@@ -63,19 +63,18 @@ func validateAuthConfig(c *AuthConfig, fldPath *field.Path) field.ErrorList {
 
 // validateLoadBalancerConfig provides basic validation of LoadBalancerConfig
 // instances.
-func validateLoadBalancerConfig(c *LoadBalancerConfig, fldPath *field.Path) field.ErrorList {
+func validateLoadBalancerConfig(c *Config, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if c == nil {
+	lbConfig := c.LoadBalancer
+	if &lbConfig == nil {
 		return append(allErrs, field.Required(fldPath, ""))
 	}
-	if c.Subnet1 == "" {
-		allErrs = append(allErrs, field.Required(fldPath.Child("subnet1"), ""))
-	}
-	if c.Subnet2 == "" {
-		allErrs = append(allErrs, field.Required(fldPath.Child("subnet2"), ""))
+	if (lbConfig.Subnet1 == "" || lbConfig.Subnet2 == "") && c.VCNID == "" {
```

Comment:
We only ever try to extract the subnet from subnet1 so you "could" simplify this logic to just checking the first subnet. 

```
if (lbConfig.Subnet1 == "") && c.VCNID == "" {
```